Ginger/GingerCoreNET/RunLib/DynamicExecutionLib/DynamicExecutionManager.cs

```diff
@@ -948,48 +997,113 @@ public static void CreateUpdateRunSetFromJSON(RunsetExecutor runsetExecutor, Gin
+                        if (runsetOperationConfigMail.MailSettings.MailCC != null)
                         {
                             mailOperation.MailCC = runsetOperationConfigMail.MailSettings.MailCC;
                         }
-                        if (string.IsNullOrEmpty(runsetOperationConfigMail.MailSettings.Subject) == false)
+                        if (runsetOperationConfigMail.MailSettings.Subject != null)
                         {
                             mailOperation.Subject = runsetOperationConfigMail.MailSettings.Subject;
                             mailOperation.Email.Subject = runsetOperationConfigMail.MailSettings.Subject;
                         }
 
-                        if (string.IsNullOrEmpty(runsetOperationConfigMail.Comments) == false)
+                        //report mail content
+                        if (runsetOperationConfigMail.Comments != null)
                         {
                             mailOperation.Comments = runsetOperationConfigMail.Comments;
                         }
+                        if (runsetOperationConfigMail.BodyContentType != null)
+                        {
+                            switch (runsetOperationConfigMail.BodyContentType)
+                            {
+                                case MailReportOperationExecConfig.eBodyContentType.HTMLReport:
+                                    mailOperation.HTMLReportTemplate = RunSetActionHTMLReportSendEmail.eHTMLReportTemplate.HTMLReport;
+                                    break;
+                                case MailReportOperationExecConfig.eBodyContentType.FreeText:
+                                    mailOperation.HTMLReportTemplate = RunSetActionHTMLReportSendEmail.eHTMLReportTemplate.FreeText;
+                                    break;
+                            }
+                        }
+                        if (runsetOperationConfigMail.BodyReportTemplateID != null)
+                        {
+                            mailOperation.selectedHTMLReportTemplateID = (int)runsetOperationConfigMail.BodyReportTemplateID;
+                        }
+                        if (runsetOperationConfigMail.BodyTextContent != null)
+                        {
+                            mailOperation.Bodytext = runsetOperationConfigMail.BodyTextContent;
+                        }
+
+                        //report attachments
+                        EmailAttachment reportAttachment = mailOperation.EmailAttachments.Where(x => x.AttachmentType == EmailAttachment.eAttachmentType.Report).FirstOrDefault();
                         if (runsetOperationConfigMail.IncludeAttachmentReport != null)
                         {
-                            if (runsetOperationConfigMail.IncludeAttachmentReport == true)
+                            if (runsetOperationConfigMail.IncludeAttachmentReport == true && reportAttachment == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5700269)